packages/compass-saved-aggregations-queries/src/hooks/use-grid-filters.tsx

```diff
@@ -0,0 +1,278 @@
+        aria-label="Search"
+        type="search"
+        placeholder="Search"
+        value={search}
+        defaultValue={search}
+        onChange={(e) => {
+          setSearch(e.target.value);
+        }}
+        spellCheck={false}
+      />
+    );
+  }, [search]);
+
+  return [searchControls, search];
+}
+
+function useSelectFilter(tree: Tree[]): [React.ReactElement, SelectState] {
+  const [selectState, setSelectState] = useState<SelectState>({});
+  const [collections, setCollections] = useState<Tree[]>([]);
+
+  const selectDatabase = useMemo(() => {
+    return (database: string): void => {
+      setSelectState({
+        database: database ?? undefined,
+      });
+      const items = tree.find((x) => x.value === database)?.items;
+      setCollections(items || []);
+    };
+  }, [tree]);
+
+  const selectCollection = useMemo(() => {
+    return (collection: string): void => {
+      setSelectState({
+        ...selectState,
+        collection: collection ?? undefined,
+      });
+    };
+  }, [selectState]);
+
+  const treeControls = useMemo(() => {
+    return (
+      <div className={selectContainer}>
+        <RenderFilterSelect
+          items={tree}
+          placeHolder={'All databases'}
+          onSelect={selectDatabase}
+          defaultValue={selectState.database}
+        />
+        <RenderFilterSelect
+          items={collections}
+          placeHolder={'All collections'}
+          onSelect={selectCollection}
+          defaultValue={selectState.collection}
+        />
+      </div>
+    );
+  }, [selectDatabase, tree, collections, selectCollection, selectState]);
+
+  return [treeControls, selectState];
+}
+
+function convertItemsToTree(items: Item[]): Tree[] {
+  const dbCollectionMap: Record<string, string[]> = {};
+  const tree = [];
+  items.forEach((item) => {
+    if (!dbCollectionMap[item.database]) {
+      dbCollectionMap[item.database] = [];
+    }
+    dbCollectionMap[item.database].push(item.collection);
+  });
+  for (const database in dbCollectionMap) {
+    const collections = dbCollectionMap[database]
+      .filter((collection, index, arr) => arr.indexOf(collection) === index)
+      .map((collection) => ({
+        value: collection,
+      }));
+
+    tree.push({
+      value: database,
+      items: collections,
+    });
+  }
+  return tree;
+}
+
+function getSearchableText(item: Item) {
+  let searchable: unknown[] = [item.name, item.database, item.collection];
+
+  if (item.type === 'aggregation') {
+    const stages = item.aggregation.pipeline
+      .filter((x) => x.stageOperator && x.stage)
+      .map((stage) => {
+        return {
+          [stage.stageOperator]: JSON.parse(stage.stage),
+        };
+      });
+    searchable = searchable.concat([
+      item.aggregation.namespace,
+      item.aggregation.env ?? '',
+      item.aggregation.collationString ?? '',
+      item.aggregation.isTimeSeries ? ['timeseries', 'time series'] : '',
+      ['aggregate', 'aggregation'],
```

Comment:
I think I get why these are here, but might make sense to document it somehow in the code as it doesn't look obvious at a glance